### PR TITLE
React app setup + Flask refactor

### PR DESCRIPTION
**Pull request by billzyc**
Implemented some functionalities for the front end and updated/created the appropriate backend endpoints for it.
- File download
- Retrieve file information
- Mobile photo upload
- Basic styling
- React context setup for global states
- NavBar

### CODE REVIEW

File: flask/config.py [hunk 1]
```diff
@@ -5,11 +5,12 @@
...
     SQLALCHEMY_DATABASE_URI = os.getenv('DATABASE_URL', 'postgresql:///photo2code_dev')
-    DEBUG = False
-    TESTING = False
+    DEBUG = True
+    TESTING = True
     CSRF_ENABLED = True
-    SECRET_KEY = os.urandom(24)  ##TODO: Change this!!
-
+    ##TODO: Change this!!
+    SECRET_KEY = 'app secret'
```
Comment (billzyc) [inline flask/config.py]: 100% change this when we actually deploy. Kept like this for local development and testing.

File: flask/migrations/versions/2b0a14520510_.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+# revision identifiers, used by Alembic.
+revision = '2b0a14520510'
+down_revision = '1e8a9b118c49'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
```
Comment (chelseatj3) [inline flask/migrations/versions/2b0a14520510_.py]: can we delete these comments

Comment (billzyc) [reply flask/migrations/versions/2b0a14520510_.py]: everything in the migrations folder is auto-generated. We probably should leave it as it is?

Comment (chelseatj3) [reply flask/migrations/versions/2b0a14520510_.py]: no we don't need those please adjust and end commands comments

File: .vscode/settings.json [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
+    "python.pythonPath": "/bin/python3"
```
Comment (chelseatj3) [inline .vscode/settings.json]: Should we have this file? It looks like it's a hidden one so not sure we should be committing it?

File: README.md [hunk 4]
```diff
@@ -85,3 +57,81 @@ Linux bash or Mac OS X:
...
+### `npm run eject`
+
+**Note: this is a one-way operation. Once you `eject`, you can’t go back!**
+
+If you aren’t satisfied with the build tool and configuration choices, you can `eject` at any time. This command will remove the single build dependency from your project.
+
+Instead, it will copy all the configuration files and the transitive dependencies (webpack, Babel, ESLint, etc) right into your project so you have full control over them. All of the commands except `eject` will still work, but they will point to the copied scripts so you can tweak them. At this point you’re on your own.
+
+You don’t have to ever use `eject`. The curated feature set is suitable for small and middle deployments, and you shouldn’t feel obligated to use this feature. However we understand that this tool wouldn’t be useful if you couldn’t customize it when you are ready for it.
+
```
Comment (chelseatj3) [inline README.md]: A lot of this readme doesn't apply to us so I would suggest removing it as other applications don't include it (as this is also the readme edits ticket)

File: .vscode/settings.json (hunk 3 above)
Comment (billzyc) [reply .vscode/settings.json]: my bad, it shouldn't be here

File: README.md (hunk 4 above)
Comment (billzyc) [reply README.md]: Sounds good

File: flask/app/templates/public/upload_image.html [hunk 5]
```diff
@@ -20,8 +20,8 @@ <h1>Upload an image</h1>
           </div>
           <label>Title</label>
           <input type="text" name="title" id="title" />
-          <label>language</label>
-          <input type="text" name="language" id="language" />
+          <label>extension</label>
```
Comment (chelseatj3) [inline flask/app/templates/public/upload_image.html]: perhaps this should begin with a capital letter like you have with title

File: flask/app/views.py [hunk 6]
```diff
@@ -52,21 +52,95 @@ def googleSignin():
...
+            db.session.add(new_file)
+            db.session.commit()
+            return jsonify({'message': 'Success'})
+        return jsonify({'message': 'Missing upload'}), 403
+    except:
+        return jsonify({'message': 'Invalid upload'}), 403
+
+@app.route('/delete_file', methods=['POST'])
+@authenticateToken
+def deleteFile():
```
Comment (chelseatj3) [inline flask/app/views.py:91]: Can we use snake case instead of camel case

File: flask/app/models.py [hunk 7]
```diff
@@ -17,9 +17,9 @@ def __init__(self, email, first_name, last_name):
         self.last_name = last_name
 
     def __repr__(self):
-        return f'User id: {id} email:{email}'
+        return f'User id: {self.id} email:{self.email}'
 
-    def profileMap(self):
+    def getMap(self):
```
Comment (chelseatj3) [inline flask/app/models.py]: Can we use snake case instead of camel case

File: flask/app/models.py [hunk 8]
```diff
@@ -34,12 +34,18 @@ class CodeFile(db.Model):
     date_posted = db.Column(db.DateTime, nullable=False,
                             default=datetime.utcnow)
     content = db.Column(db.Text, nullable=True)
-    language = db.Column(db.String(50), nullable=False)
+    extension = db.Column(db.String(50), nullable=False)
     # image_file = db.Column(db.String(20), nullable=False)
     user_id = db.Column(db.Integer, db.ForeignKey('user.id'), nullable=False)
+    
+    def getMap(self):
```
Comment (chelseatj3) [inline flask/app/models.py]: Same here

**Comment by chelseatj3** [comment]
Both functions in `authenticate.py` should be snake case

**Comment by chelseatj3** [comment]
Should we be committing these files `.DS_Store`?

File: flask/app/views.py [hunk 9]
```diff
@@ -52,21 +52,95 @@ def googleSignin():
...
+            return jsonify({'message': 'Success'})
+        return jsonify({'message': 'Missing user or file information'}), 403
+    except:
+        return jsonify({'message': 'Request Error'}), 403
+
 
 
+@app.route('/files', methods=['GET'])
+@authenticateToken
+def files():
```
Comment (chelseatj3) [inline flask/app/views.py]: I think this should also be renamed to say what we do which is get_files

File: .gitignore [hunk 10]
```diff
@@ -1,5 +1,36 @@
...
+
+# testing
+react-app/coverage
+
+# production
+react-app/build
+
+# misc
+react-app/.DS_Store
+react-app/.env.local
```
Comment (chelseatj3) [inline .gitignore]: can we just add `react-app/.env.*` instead of all the various .env denominations?

**Comment by chelseatj3** [comment]
nit: all the python files etc have a certain spacing, can we also continue in the js and scss files too for style consistency? Perhaps running a styling package (https://pypi.org/project/oitnb/) which takes care of the double quotes unsure of the spacing

File: react-app/src/components/FilesList/FilesList.js [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+    return fileTiles;
+  };
+  return (
+    <div className="files-list">
+      <div className="files-container">
+        {userFiles?.length > 0 ? (
+          renderFileTiles()
+        ) : (
+          <div>
+            {" "}
```
Comment (chelseatj3) [inline react-app/src/components/FilesList/FilesList.js]: what is this line for?

File: .gitignore (hunk 10 above)
Comment (billzyc) [reply .gitignore]: good idea!

**Review by chelseatj3 — APPROVED** [review]
LGTM

**Pull request merged**